src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -304,7 +313,10 @@ function VisEditor($scope, $route, timefilter, AppState, $location, kbnUrl, $tim
 
       if (id) {
         notify.info('Saved Visualization "' + savedVis.title + '"');
-        if (savedVis.id === $route.current.params.id) {
+        if ($scope.addToDashMode) {
+          const dashboardBaseUrl = chrome.getNavLinkById('kibana:dashboard');
+          window.location.href = `${dashboardBaseUrl.lastSubUrl}&${DashboardConsts.ADD_VIS_PARAM}=${savedVis.id}`;
```

Comment:
I added a comment, it's because the dashboard url is a full url. Thought it was safer to just redirect there, rather than add something to detect the right suffix to use, which would be fragile to any url prefix changes.